system/jlib/jutil.cpp

```diff
@@ -2455,25 +2455,41 @@ jlib_decl bool querySecuritySettings(DAFSConnectCfg *_connectMethod,
              if (*_privateKey)
                  dfsKeywords = true;
         }
+
+        StringBuffer DAFSpassPhraseEnc;
         if (_passPhrase)
         {
             CriticalBlock b(DAFSpassPhraseCrit);
-            if (DAFSpassPhrase.isEmpty())
+            if (DAFSpassPhraseDec.isEmpty())//previously retrieved/decrypted it?
             {
                 const char *passPhrasePtr = conf.queryProp("dfsSSLPassPhrase");
                 if (!isEmptyString(passPhrasePtr))
                 {
+                    DAFSpassPhraseEnc.append(passPhrasePtr);//got encrypted pwd
                     dfsKeywords = true;
-                    decrypt(DAFSpassPhrase, passPhrasePtr);
                 }
             }
-            *_passPhrase = DAFSpassPhrase.str();
         }
 
         if (!dfsKeywords && (_certificate || _privateKey || _passPhrase))
         {
         //end of deprecated code
-            queryHPCCPKIKeyFiles(_certificate, _privateKey, _passPhrase);//use new keywords
+            const char *passPhrasePtr = nullptr;
+            queryHPCCPKIKeyFiles(_certificate, _privateKey, _passPhrase ? &passPhrasePtr : nullptr);//use new keywords
```

Comment:
I think in this querySecuritySettings() call a certificate file is expected and used for SSL handshake and authentication.  So the overlap of querySecuritySettings() and queryHPCCPKIKeyFiles() is not exact.